kubeflow/trainer/rhai/traininghub.py

```
@@ -573,6 +577,110 @@ def _transform_lora_sft(self, metrics):
+
+            GRPO writes two lines per iteration (rollout + train phase).
+            Read the last 2 lines and merge them into a single dict.
+            """
+            metrics_file = f"{ckpt_output_dir}/{metrics_file_rank0}"
+
+            try:
+                if not os.path.exists(metrics_file):
+                    return {}
+
+                try:
+                    result = subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)